src/mam/mod_mam_utils.erl

```diff
@@ -677,6 +679,11 @@ is_mam_namespace(?NS_MAM_04) -> true;
 is_mam_namespace(?NS_MAM_06) -> true;
 is_mam_namespace(_)          -> false.
 
+features(Module, Host) ->
+    [?NS_MAM_04, ?NS_MAM_06] ++ case has_message_retraction(Module, Host) of
+                                    true -> [?NS_RETRACT];
+                                    false -> []
+                                end.
```

Comment:
I'm not fun (actually against) mixing results from `case` clauses with other operations. Can this block of code be changed to sth like the following?
```suggestion
    MAMFeatures = [?NS_MAM_04, ?NS_MAM_06],
    case has_message_retraction(Module, Host) of
        true -> [?NS_RETRACT | MAMFeatures];
        false -> MAMFeatures
    end.
```

Reply:
Yes. The only question is: can `?NS_RETRACT` stay at the end? I think it's good if the list starts from the MAM features.

Reply:
Sure, it can stay at the end.